crates/modelardb_server/src/storage/uncompressed_data_buffer.rs

```diff
@@ -701,6 +627,78 @@ mod tests {
     }
     }
 
+    #[tokio::test]
+    async fn test_get_on_disk_data_buffer_disk_size() {
+        let temp_dir = tempfile::tempdir().unwrap();
+
+        let uncompressed_on_disk_buffer = create_on_disk_data_buffer(temp_dir.path()).await;
+
+        assert_eq!(uncompressed_on_disk_buffer.disk_size(), 3135)
+    }
+
+    #[tokio::test]
+    async fn test_check_if_on_disk_data_buffer_is_unused() {
+        let temp_dir = tempfile::tempdir().unwrap();
+
+        let uncompressed_on_disk_buffer = create_on_disk_data_buffer(temp_dir.path()).await;
+
+        assert!(!uncompressed_on_disk_buffer.is_unused(CURRENT_BATCH_INDEX));
+        assert!(uncompressed_on_disk_buffer.is_unused(CURRENT_BATCH_INDEX + 1));
+    }
+
+    #[tokio::test]
+    async fn test_read_in_memory_data_buffer_from_on_disk_data_buffer() {
+        let temp_dir = tempfile::tempdir().unwrap();
+
+        let uncompressed_on_disk_buffer = create_on_disk_data_buffer(temp_dir.path()).await;
+
+        let read_uncompressed_in_memory_buffer = uncompressed_on_disk_buffer
+            .read_from_apache_parquet(CURRENT_BATCH_INDEX)
+            .await
+            .unwrap();
+
+        // The creation of record_batch empties uncompressed_in_memory_buffer_to_be_spilled.
+        let mut uncompressed_in_memory_buffer = UncompressedInMemoryDataBuffer::new(
+            UNIVARIATE_ID,
+            test::model_table_metadata_arc(),
+            CURRENT_BATCH_INDEX,
+        );
+
+        insert_data_points(
+            uncompressed_in_memory_buffer.capacity(),
+            &mut uncompressed_in_memory_buffer,
+        );
+
+        assert_eq!(
+            uncompressed_in_memory_buffer.timestamps.values_slice(),
+            read_uncompressed_in_memory_buffer.timestamps.values_slice()
+        );
+
+        assert_eq!(
+            uncompressed_in_memory_buffer.values.values_slice(),
+            read_uncompressed_in_memory_buffer.values.values_slice()
+        );
+    }
+
+    /// Create an on-disk data buffer at `path` from a full `UncompressedInMemoryDataBuffer`.
```

Comment:
Maybe change `path` to match the argument name.

Reply:
Fixed.